modules/minimp3/register_types.cpp

```diff
@@ -0,0 +1,52 @@
+/*************************************************************************/
+/*  register_types.cpp                                                   */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md)    */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "register_types.h"
+
+#include "audio_stream_mp3.h"
+
+#ifdef TOOLS_ENABLED
+#include "core/engine.h"
+#include "resource_importer_mp3.h"
+#endif
+
+void register_minimp3_types() {
+#ifdef TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		Ref<ResourceImporterMP3> mp3_import;
+		mp3_import.instance();
+		ResourceFormatImporter::get_singleton()->add_importer(mp3_import);
+	}
```

Comment:
The editor check is not needed because the register code already wrapped in `TOOLS_ENABLED`.

```suggestion
	Ref<ResourceImporterMP3> mp3_import;
	mp3_import.instance();
	ResourceFormatImporter::get_singleton()->add_importer(mp3_import);
```

Reply:
`TOOLS_ENABLED` means that it's an editor build, but not necessarily that you're running the editor. So checking the editor hint (to check that this is an editor *instance* is correct). The same is done for OGG Vorbis.

Reply:
Is this specific to audio importers or importers in general? Because most other existing importers don't do this check. 😮

Anyways, also talking about godotengine/godot-proposals#1632.

Perhaps a better way to do this is how `assimp` module registers init callback:

https://github.com/godotengine/godot/blob/f0f8864c9c588c2d7f012dc76f5451d9df5dd3b2/modules/assimp/register_types.cpp#L36-L55

Reply:
See also godotengine/godot-proposals#1362, so I'm not sure whether it's feasible to restrict this to editor instance.